### DISCUSSION TITLE
Semester 4 Woche 9

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#54] [Student Exchange]
Nachdem Ihr ja jetzt lange nichts mehr von uns gehört hattet,
freuen wir uns mal wieder von unserem 💹 Fortschritt berichten zu können.

Die letzten Wochen haben wir einige verbliebenen Features für die Demo implementiert.
Darunter sind die Events sowie die 🖱️ Mouse und ⌨️ Key Listener.
Diese bieten in der Engine nun 🌐 Schnittstellen, um das Verhalten frei zu programmieren.

Wir haben uns auch sehr stark 💪 um das Linting des Projektes gekümmert.
Dabei haben wir von anfänglichen 2000 ❌ Fehlern nun nur noch knapp 400.
Somit haben wir 80% der Fehler bereits behoben 📉 , dadurch liegt unser momentaner Messwert bei **4 Fehlern pro 100 Zeilen** Code

Des weiteren haben wir diese Woche im Team eine ⏪ Retrospektive durchgeführt.
Das Ganze haben wir auf folgendem 📋 Concept Board fest gehalten:

![Screenshot Concept Board](https://github.com/Christian-2003/CrInGE/assets/136824971/34496a65-7dda-4b21-bbdf-9fe47c59cb3e)

Das wäre dann auch schon alles für diese Wochen.
Im nächsten Blog werden wir nochmals alle unseren bis dahin erreichten 💹 Fortschritte und 📚 Dokumente zusammenfassen.

**Comment by davidlzt** [comment]
Hallo,
Das war ein guter Beitrag. Viele gute Aspekte dabei 😄 
Wir freuen uns aufs nächste Mal!
